FundTrack.WebUI/Controllers/ReportsController.cs

```diff
@@ -22,21 +23,58 @@ public ReportsController(IReportService service, IOrganizationStatisticsService
         }
 
         [HttpGet("IncomeReport")]
-        public IEnumerable<ReportIncomeViewModel> GetIncomeReports(int orgId, DateTime? dateFrom, DateTime? dateTo)
-        {
-            return _service.GetIncomeReports(orgId, dateFrom, dateTo);
+        public ActionResult GetIncomeReports(int orgId, DateTime? dateFrom, DateTime? dateTo)
+        { 
+            if (isDateValid(dateFrom, dateTo) && isIdValid(orgId))
+            {
+                return Ok(_service.GetIncomeReports(orgId, dateFrom, dateTo));
+            }           
+                return BadRequest(string.Format(ErrorMessages.IncomeReportErrorMessage, orgId, dateFrom, dateTo));     
         }
 
         [HttpGet("OutcomeReport")]
-        public IEnumerable<ReportOutcomeViewModel> GetOutcomeReports(int orgId, DateTime? dateFrom, DateTime? dateTo)
+        public ActionResult GetOutcomeReports(int orgId, DateTime? dateFrom, DateTime? dateTo)
         {
-            return _service.GetOutcomeReports(orgId, dateFrom, dateTo);
+            if (isDateValid(dateFrom, dateTo) && isIdValid(orgId))
+            {
+                return Ok(_service.GetOutcomeReports(orgId, dateFrom, dateTo));
+            }
+            return BadRequest(string.Format(ErrorMessages.OutcomeReportErrorMessage, orgId, dateFrom, dateTo));

```

Comment:
use ModelState object (inherited property from Controller base class) instead of string.Format(ErrorMessages.FinopImagesErrorMessage, finOpId